mern-ecommerce/controllers/product.js

```diff
@@ -8,9 +8,9 @@ exports.productById = (req, res, next, id) => {
   Product.findById(id)
     .populate('category')
     .exec((err, product) => {
-      if (err || !product) {
+      if (!err || !product) {
         return res.status(400).json({
-          error: 'Product not found',
+          erroar: 'Product not found',
```

Comment:
"Typo: 'erroar' should be 'error'. It's a common mistake, but it's essential to maintain consistency in error messages."